[docs] Add note about expanding query planner heap size (#5342)
apollographql/router: docs/source/configuration/overview.mdx

Learn how to customize the behavior of your Apollo Router with environment variables, command-line commands and options, and YAML file configuration.

## Environment variables

> Most environment variables are also documented with their [Command-line options](#command-line-options) counterpart. However, there are some options that are only available to configure as environment variables.

Reviewer: This addition confuses me a bit since I don't see the CLI counterpart in the table below.

Also, we should specify which options are only available via env variable.

Author: > I don't see the CLI counterpart in the table below.

That is in the next section below `## Command-line options`

> Also, we should specify which options are only available via env variable.

That is supposed to be this section here.

Reviewer: Got it, would suggest this to clarify then:

```suggestion

You can set most environment variables via their [command-line option](#command-line-options) counterpart. The following values are only configureable as environment variables:
```

    </tr>
  </thead>

  <tbody>
    <tr class="required">
      <td style="min-width: 150px;">
      
      ##### `APOLLO_GRAPH_REF`
      
      </td>
      <td>
      
      The graph ref for the GraphOS graph and variant that the router fetches its supergraph schema from (e.g., `docs-example-graph@staging`).
      
      **Required** when using [managed federation](/federation/managed-federation/overview/), except when using an [offline license](#--license) to run the router.
      
      </td>
    </tr>
    <tr class="required">
      <td style="min-width: 150px;">
      
      ##### `APOLLO_KEY`
      
      </td>
      <td>
      
      The [graph API key](/graphos/api-keys/#graph-api-keys) that the router should use to authenticate with GraphOS when fetching its supergraph schema.
      
      **Required** when using [managed federation](/federation/managed-federation/overview/), except when using an [offline license](#--license) to run the router.
      
      </td>
    </tr>
    <tr>
      <td style="min-width: 150px;">
      
      ##### `APOLLO_ROUTER_BRIDGE_EXPERIMENTAL_V8_INITIAL_HEAP_SIZE`
      
      </td>
      <td>

>       `EXPERIMENTAL`: The starting size of query planner heap to use in MB. Setting `APOLLO_ROUTER_BRIDGE_EXPERIMENTAL_V8_INITIAL_HEAP_SIZE=4096` is ~4GB.

Reviewer: ```suggestion
      The initial size to use for query planner heaps in MB. For example, setting this variable to `4096` sets an initial heap size of ~4GB.
```

>       Apollo Router v1.x uses a JavaScript runtime internally to run the query planner. While this is mostly an internal implementation detail, large supergraphs tend to increase the amount of memory required in the Router. The query planner heap will dynamically expand as needed however, increasing the starting size of the memory allocated to the heap can improve the first planning times for large operations against a large supergraphs.

Reviewer: ```suggestion
      The Apollo Router v1.x uses a JavaScript runtime internally to run the query planner. Large supergraphs tend to increase the amount of memory required in the router. The query planner heap dynamically expands as needed but increasing the initial size of the memory allocated to the heap can improve the performance of initial large operations against large supergraphs.
```
